redaxo/src/addons/phpmailer/lib/mailer.php

```diff
@@ -52,13 +57,58 @@ public function __construct($exceptions = false)
         parent::__construct($exceptions);
     }
 
+    /**
+     * Override the AddAddress method of Parent
+     * Same Parameters as Parent
+     *
+     * $address: string / Adress of the reciever / mail_to
+     * $name: string / Name of the reciever / mail_to_name
+     *
+     * @author markus[dot]dick[at]novinet[dot]de Markus Dick
+     *
+    */
+    public function AddAddress($address, $name = '') {
+        $addon = rex_addon::get('phpmailer');
+        
+        // checks if detour mode is enabled
+        // $detour: boolean
+        $detour = true == $addon->getConfig('detour_mode') && '' != $addon->getConfig('test_address');
+
+        // sets the address to the detour address
+        if(true == $detour) {
+            $detour_address = $addon->getConfig('test_address');
+            // check if address is valid
+            if(true == rex_validator::factory()->email($detour_address)) {
+                // store the address so we can use it in the subject later
+                $this->original_mail_to = $address;
+
+                // Set $address to the detour address
+                $address = $detour_address;
+            }
+        } 
+        
+        // use the parents method to add the address
+        // wether its the original address or the detour address
+        return parent::addAddress($address, $name);
+    }
+
     public function send()
     {
         return rex_timer::measure(__METHOD__, function () {
             if ($this->archive) {
                 $this->archive();
             }
             $addon = rex_addon::get('phpmailer');
+
+            // checks if detour mode is enabled
+            // $detour: boolean
+            $detour = true == $addon->getConfig('detour_mode') && '' != $addon->getConfig('test_address');
+
+            // Sets Subject of E-Mail to [DETOUR] $subject [$original_mail_to]
+            if(true == $detour &&  '' != $this->original_mail_to) {
+                $this->Subject = '[DETOUR] ' . $this->Subject . ' [originally for: ' . $this->original_mail_to . ']';
```

Comment:
Subject bitte in der default Backend-Sprache ausliefern.

Reply:
wird erledigt

Reply:
ist erledigt